src/mixins/java/org/spongepowered/common/mixin/core/server/level/ServerLevelMixin.java

```diff
@@ -443,7 +444,7 @@ public void save(@Nullable final ProgressListener progress, final boolean flush,
 
     @Inject(method = "tick", at = @At("RETURN"))
     private void impl$unloadBlockEntities(final BooleanSupplier param0, final CallbackInfo ci) {
-        if (!this.blockEntitiesToUnload.isEmpty()) {
+        if (this.emptyTime >= 300 && !this.blockEntitiesToUnload.isEmpty()) {
```

Comment:
No, this isn't what I meant.

> Could we perhaps mix into the this.emptyTime++ < 300 boolean check to do it then? 

i.e. into the `if (active || this.emptyTime++ < 300) {` line, probably `PUTFIELD` on `this.emptyTime`.